src/util.hpp

```diff
@@ -19,33 +19,38 @@
 #include <sstream>
 #include <string>
 #include <vector>
+#include <type_traits>
 
 namespace bls {
 
 class BLS;
 
-class Bytes {
-    const uint8_t* pData;
-    const size_t nSize;
+template <typename T>
+class span {
+    T* pData;
+    size_t nSize;
 
 public:
-    explicit Bytes(const uint8_t* pDataIn, const size_t nSizeIn)
+    explicit span(const T* pDataIn, const size_t nSizeIn)
         : pData(pDataIn), nSize(nSizeIn)
     {
     }
-    explicit Bytes(const std::vector<uint8_t>& vecBytes)
-        : pData(vecBytes.data()), nSize(vecBytes.size())
+
+    span(const std::vector<typename std::remove_const<T>::type>& cont)
```

Comment:
note how `explicit` was removed here. This allows implicit conversion from a vector to a span. This lets you pass either a vector or a pointer+size to an interface that just wants a sequence of objects.